checkout-ui-custom/checkout6-custom.js

```diff
@@ -444,7 +444,7 @@ const MAX_TIME_EXPIRATION = 1000 * 60 * 5 // 5 minutes
         isWorkspace() ? `?v=${ts}` : ''
       }`,
     }).then(function (response) {
-      if (Object.keys(response).length === 0) {
+      if (Object.keys(response).length === 0 || response.error === 'User not authenticated') {
```

Comment:
Did you simulate this error? I requested the endpoint without a token or cookie and the response was successful (http 200), it didn't return the error.

Reply:
Hi @Rudge 
Please test on the public host of the account, not on .myvtex.com

Reply:
![Screenshot 2023-09-14 at 12 28 00](https://github.com/vtex-apps/b2b-checkout-settings/assets/84322200/a8877a6d-a962-4479-8210-cd7f2fe531cc)
I can get the error when trying on the public host.

Reply:
For a custom domain, maybe the problem is elsewhere. I found this [condition `!host?.includes('myvtex.com')`](https://github.com/vtex-apps/b2b-checkout-settings/blob/master/node/resolvers/Routes/index.ts#L18), can you test removing it?

Reply:
Sorry for my mistake in understanding the problem. I believe that the best convention should be to handle the HTTP status `401` and replace the if catching the error.

Endpoint
```Javascript
if (!token) {
      ctx.response.body = {
        error: 'User not authenticated',
      }
      ctx.response.status = 401

      ctx.set('cache-control', 'no-cache')
    }
```

checkout6-custom
```Javascript
$.ajax({
      url: `${rootPath}/_v/private/b2b-checkout-settings/${
        isWorkspace() ? `?v=${ts}` : ''
      }`,
     error: function(err) {
         window.sessionStorage.removeItem('b2b-checkout-settings')

        return
    }
```

What do you think?

Reply:
Hi @Rudge 
Code 401 forces you to log in...
403 (forbidden) which would be more suitable for our scenario also displays a modal in which it informs the client that it does not have permissions in order not to have side-effects, we should use something generic, such as 400

Reply:
I suggested the 401 because the token doesn't exist and it is necessary to login and have the token. But ok, for me, I agree with 400 or 403 too.

Reply:
Hi @Rudge 
I made another commit with your suggestions.

Please review again.

Thank you!